### PR TITLE
Some code cleanup + add a missing `@inbounds`

### PR DESCRIPTION
**Pull request by navidcy**
Most changes are aesthetic and less code.

Only adding a few `@inbounds` is possibly the important changes.

### CODE REVIEW

**Comment by tomchor** [comment]
Thanks! I wonder how much difference the `@inbounds` makes...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
       fail-fast: false
       matrix:
         version:
-          - '1.8'
+          - '1.9'
```
Comment (tomchor) [inline .github/workflows/ci.yml:12]: I'm reluctant to test a Julia version that Oceananigans isn't tested against yet, and not test the version that Oceananigans _is_ tested agaisnt.

Comment (navidcy) [reply .github/workflows/ci.yml:12]: Damn! That was a mistake! I was meant to do that for a different repo and did it here apparently... I'll remove this! :)

**Review by tomchor — APPROVED** [review]
Thanks! This looks great

**Pull request merged**